### PR TITLE
Upgrade sendgrid/sendgrid to 5, remove Unirest, update base code

### PR DESCRIPTION
**Pull request by cristianooliveira**
Hello,

I made this pull request to upgrade the base sendgrid/sendgrid package and the base code. With this and since sendgrid removed Unirest, i also removed it. 

All methods will work in the same way but configs were changed since sendgrid changes the way to connect with their api, now we use a API_KEY instead of user - password to make requests.
This change is related with the new version of sendgrid api.

### CODE REVIEW

File: Service/SendGridTemplatingMailerService.php [hunk 1]
```diff
@@ -144,40 +149,68 @@ protected function processResponse(StdClass $response, $throwExceptionOnFail = t
...
+                $attachment = new Attachment();
+                $attachment->setContent(base64_encode(file_get_contents($attachmentPath)));
+                $attachment->setFilename(basename($attachmentPath));
+                $attachment->setDisposition("attachment");
+                $email->addAttachment($attachment);
+            }
         }
 
+        $email->addPersonalization($personalization);
+        $email->addPersonalization($personalization);
```
Comment (seeruk) [inline Service/SendGridTemplatingMailerService.php]: Should this be here twice?

File: composer.json [hunk 2]
```diff
@@ -14,8 +14,7 @@
     "keywords": ["sendgrid", "mail", "email", "web service", "smtp"],
     "require": {
         "symfony/templating": ">=2.4.1",
-        "mashape/unirest-php": "dev-master",
```
Comment (seeruk) [inline composer.json:17]: Glad this is going!

File: Service/SendGridTemplatingMailerService.php (hunk 1 above)
Comment (cristianooliveira) [reply Service/SendGridTemplatingMailerService.php]: Nope ;)

**Comment by seeruk** [comment]
LGTM, this should probably be a v2 release.

**Comment by cristianooliveira** [comment]
Ok, thanks for the review!

**Comment by eduardoconceicao** [comment]
+1

**Review by eduardoconceicao — APPROVED** [review]

**Comment by eduardoconceicao** [comment]
@SeerUK Can we get this merged?

**Comment by seeruk** [comment]
Done, and released as v2.0.0 :)

**Comment by eduardoconceicao** [comment]
Nice one! Thanks

**Pull request merged**